crates/core/types/account.rs

```diff
@@ -108,26 +108,28 @@ impl RLPDecode for AccountState {
     }
 }
 
-pub fn compute_storage_root(storage: &HashMap<H256, H256>) -> H256 {
-    let rlp_storage = storage
-        .iter()
-        .map(|(k, v)| {
-            let mut k_buf = vec![];
-            let mut v_buf = vec![];
-            k.encode(&mut k_buf);
-            v.encode(&mut v_buf);
-            (k_buf, v_buf)
-        })
-        .collect::<Vec<_>>();
-    let root =
-        PatriciaMerkleTree::<_, _, Keccak256>::compute_hash_from_sorted_iter(rlp_storage.iter());
+pub fn compute_storage_root(storage: &BTreeMap<H256, H256>) -> H256 {
+    let mut pmt = PatriciaMerkleTree::<Vec<u8>, Vec<u8>, Keccak256>::new();
+
+    for (k, v) in storage.iter() {
+        let mut v_buf = vec![];
+        let k_buf = Keccak256::new_with_prefix(k).finalize().to_vec();
+        v.encode(&mut v_buf);
+        pmt.insert(k_buf, v_buf);
```

Comment:
you were saying that if you inserted key value pairs in different order then the root would have been different?

Reply:
Not really. See https://github.com/lambdaclass/ethereum_rust/issues/203.

There is an utility function to calculate root hash without actually building the trie, provided the list is sorted

Reply:
It was done [here](https://github.com/lambdaclass/ethereum_rust/pull/183/files#diff-c79f14c01fe86e1edb154b9b6a79a31c193034a2cca218af26f5143c4e220913L123) and calculating an incorrect storage_root because the key hashes were not sorted.